src/modules/decklink/consumer/vanc_op47_strategy.cpp

```diff
@@ -0,0 +1,130 @@
+#include "../util/base64.hpp"
+#include "vanc.h"
+#include <bits/stdc++.h>
+#include <boost/lexical_cast.hpp>
+#include <common/param.h>
+#include <common/ptree.h>
+#include <mutex>
+#include <queue>
+#include <vector>
+
+namespace caspar { namespace decklink {
+
+const uint8_t OP47_DID  = 0x43;
+const uint8_t OP47_SDID = 0x02;
+
+class vanc_op47_strategy : public decklink_vanc_strategy
+{
+  private:
+    static const std::wstring Name;
+
+    std::mutex mutex_;
+    uint8_t    line_number_;
+    uint8_t    line_number_2_;
+    uint8_t    sd_line_;
+    uint16_t   counter_;
+
+    std::vector<uint8_t>             dummy_header_;
+    std::queue<std::vector<uint8_t>> packets_;
+
+  public:
+    explicit vanc_op47_strategy(uint8_t line_number, uint8_t line_number_2, const std::wstring& dummy_header)
+        : line_number_(line_number)
+        , line_number_2_(line_number_2)
+        , sd_line_(21)
+        , counter_(1)
+        , dummy_header_(dummy_header.empty() ? std::vector<uint8_t>() : base64_decode(dummy_header))
+    {
+    }
+
+    virtual bool        has_data() const { return !packets_.empty() || !dummy_header_.empty(); }
+    virtual vanc_packet pop_packet(bool field2)
```

Comment:
```suggestion
    virtual bool        has_data() const override { return !packets_.empty() || !dummy_header_.empty(); }
    virtual vanc_packet pop_packet(bool field2) override
```
missing `override` for all virtual methods here, also in `vanc_scte104_stategy.cpp`

Reply:
oh, also you can't read from `packets_` without locking `mutex_`.